docs/sequence_diagrams/registration.md

```diff
@@ -0,0 +1,225 @@
+Once a third-party software (e.g., CC Intel Platform Registration Service) reads the platform manifest, the BIOS expects a signal to stop sharing that manifest as it contains sensitive information (i.e., the platform keys).
+
+Given a platform manifest, one can then register the platform with the Intel Registration Service, which stores the platform keys (through the direct registration [2]) to process further queries (e.g., to retrieve a Provisioning Certification Key (PCK) Certificate [3]).
+This has the advantage that we don't need to handle the persistence of the platform manifests; we outsource that to Intel.
+One can check if the direct registration was performed by checking the `Key Caching Policy` property of a PCK Cert [3].
+
+To retrieve a PCK Certificate, necessary to carry out a successful attestation procedure by verifying quotes, one has to retrieve two pieces of information:
+(1) the Provisioning Certification Enclave (PCE) ID; and 
+(2) an encrypted PPID which is derived by the PCE from the platform keys.
+The PCE is an Intel SGX architectural enclave that uses a PCK to sign REPORT structures for Provisioning or Quoting Enclaves.
+These signed REPORTS contain the ReportData indicating that attestation keys or provisioning protocol messages are created on genuine SGX hardware [2].
+
+The outcome of a successful platform registration with Intel is the capability to retrieve the platform's PCK Certificate from the Intel Platform Certification Service (PCS).
+
+[^1]: *shared* is relevant in the context of multi-package platforms (i.e., multiple CPUs) where the CPUs negotiate the platform key to use.
+
+## Limitation
+
+- Addition or replacement of CPUs is not supported.
+
+## Status Code
+
+The platform registration service keeps a status code described below.
+
+### Prometheus Implementation
+
+To implement the status code above, we use [Prometheus Gauge Metric](https://prometheus.io/docs/concepts/metric_types/#gauge) `service_status_code`, and attach to it two [labels](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels): 
+- `http_status_code`: to represent the HTTP status code returned by the performed HTTP request
+- `intel_error_code`: a string, e.g. `InvalidRequestSyntax`, taken as-is, when available, from the Intel service reply's error code 
+
+### Semantics
+
+- `0X`: SGX status
+  - `00`: Pending execution
+  - `01`: SGX UEFI variables not available 
+  - `02`: Impossible to determine the registration status; please reattempt
+    - MIGHT contain metric label `http_status_code`
+    - MIGHT contain metric label `intel_error_code`
+  - `03`: Impossible to determine the registration status; please reset the SGX
+    - MUST contain label `http_status_code`
+  - `04`: Failed to persist the UEFI variable content
+  - `05`: Platform registered successfully and a reboot is required
+  - `09`: Platform directly registered
+- `1X`: Registration Status
```

Comment:
Currently the category reflects error cases only and is (most likely) bound to the http request status:

```suggestion
- `1X`: HTTP request Status
```

What do you think?

Reply:
Updated. Resolving thread.